source/opt/scalar_replacement_pass.cpp

```diff
@@ -144,6 +162,55 @@ Pass::Status ScalarReplacementPass::ReplaceVariable(
   return Status::SuccessWithChange;
 }
 
+bool ScalarReplacementPass::ReplaceWholeDebugDeclare(
+    Instruction* dbg_decl, const std::vector<Instruction*>& replacements) {
+  // Insert Deref operation to the front of the operation list of |dbg_decl|.
+  Instruction* dbg_expr = context()->get_def_use_mgr()->GetDef(
+      dbg_decl->GetSingleWordOperand(kDebugValueOperandExpressionIndex));
+  auto* deref_expr =
+      context()->get_debug_info_mgr()->DerefDebugExpression(dbg_expr);
+
+  // Add DebugValue instruction with Indexes operand and Deref operation.
+  int32_t idx = 0;
+  for (const auto* var : replacements) {
+    Instruction* added_dbg_value =
+        context()->get_debug_info_mgr()->AddDebugValueWithIndex(
+            dbg_decl->GetSingleWordOperand(
+                kDebugDeclareOperandLocalVariableIndex),
+            var->result_id(), deref_expr->result_id(),
+            context()->get_constant_mgr()->GetSIntConst(idx), var->NextNode());
```

Comment:
This is really hard to ready.  Each parameter is long, and it is not clear what each one is.  Could you find a way to make this clearer?  Maybe create some local variables that will be passed in the parameters.

Reply:
done